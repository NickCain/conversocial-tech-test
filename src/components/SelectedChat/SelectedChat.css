.Selected-chat {
    position: relative;
    flex-grow: 4;
}
