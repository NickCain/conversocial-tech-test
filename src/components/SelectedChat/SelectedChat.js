import React, { Component } from 'react';

import TextBox from  '../../components/TextBox/TextBox';
import ChatMessage from '../../components/ChatMessage/ChatMessage';

import './SelectedChat.css';

class SelectedChat extends Component {
  render() {
    const { onSendMessage, message } = this.props;
    return (
      <div className="Selected-chat">
        <h3>Selected Chat</h3>

        <ChatMessage type="app" message='app'/>

        <ChatMessage type="user" message='user'/>

        <TextBox onSendMessage={onSendMessage}/>
      </div>
    );
  }
}

export default SelectedChat;
