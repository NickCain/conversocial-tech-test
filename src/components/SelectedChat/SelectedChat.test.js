import { mount, shallow } from 'enzyme';
import React from 'react';
import SelectedChat from './SelectedChat.js'

let wrapper;
let shallowWrapper;

describe('SelectedChat - Component', () => {
  describe('SelectedChat renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <SelectedChat />
      );
      shallowWrapper = shallow(
        <SelectedChat />
      );
    });

    it('should render the SelectedChat component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });
  });
});
