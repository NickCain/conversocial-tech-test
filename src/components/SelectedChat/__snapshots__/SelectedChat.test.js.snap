// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedChat - Component SelectedChat renders should render the SelectedChat component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Selected-chat"
>
    <h3>
        Selected Chat
    </h3>
    <ChatMessage
        message="welcome to the chat"
        type="app"
    />
    <TextBox
        onSendMessage={undefined}
    />
</div>,
  "nodes": Array [
    <div
      className="Selected-chat"
>
      <h3>
            Selected Chat
      </h3>
      <ChatMessage
            message="welcome to the chat"
            type="app"
      />
      <TextBox
            onSendMessage={undefined}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SelectedChat />,
      "_debugID": 12,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SelectedChat {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="Selected-chat"
>
          <h3>
                    Selected Chat
          </h3>
          <ChatMessage
                    message="welcome to the chat"
                    type="app"
          />
          <TextBox
                    onSendMessage={undefined}
          />
</div>,
        "_debugID": 13,
        "_renderedOutput": <div
          className="Selected-chat"
>
          <h3>
                    Selected Chat
          </h3>
          <ChatMessage
                    message="welcome to the chat"
                    type="app"
          />
          <TextBox
                    onSendMessage={undefined}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SelectedChat />,
}
`;
